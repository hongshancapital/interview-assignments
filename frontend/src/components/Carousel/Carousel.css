html{
  font-size: 100px;
}
.wrapper {
  height: 100vh;
  overflow: hidden;
  position: relative;
}
.imgList {
  display: flex;
  width: 100%;
  height: 100%;
  transition: all 0.8s ease;
}
.imgItem {
  width: 100%;
  flex: none;
  height: 100%;
  background-size: auto 100%;
  background-position-y: bottom;
  background-position-x: center;
  background-repeat: no-repeat;
}
.imgTitle {
  font-size: 0.6rem;
  padding: 1rem 0.3rem 0.2rem 0.3rem;
  margin: 0 auto;
}
.imgDes {
  font-size: 0.3rem;
  padding: 0rem 0.3rem 0 0.3rem;
  margin: 0 auto;
}
.dotList {
  position: fixed;
  bottom: 10px;
  left: 50%;
  display: flex;
  transform: translate(-50%, -50%);
}
.dotItem {
  width: 70px;
  height: 3px;
  background-color: #a9a9a9;
  border-radius: 3px;
  margin: 5px;
  cursor: pointer;
}
.inner {
  height: 100%;
  border-radius: 3px;
}
@keyframes progress {
  from {
    width: 0px;
  }
  to {
    width: 70px;
  }
}
.line {
  width: 100%;
  background-color: #ffffff;
}
