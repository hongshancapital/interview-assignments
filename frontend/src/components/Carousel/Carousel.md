## API

| 参数            | 说明                                             | 类型                      | 默认值 |
| --------------- | ------------------------------------------------ | ------------------------- | ------ |
| autoplay        | 是否自动切换，值为数字时是轮播间隔时间 单位是 ms | boolean                   | false  |
| speed           | 切花轮播时的过渡动画时长 单位是 ms               | number                    | 300    |
| showSwitch      | 是否显示左右切换按钮                             | boolean                   | false  |
| dots            | 是否显示面板指示点                               | boolean                   | true   |
| className       | 轮播组件根元素样式名称                           | string                    |        |
| onChangeCurrent | 切换面板的回调                                   | (current: number) => void |        |

## 方法

| 名称   | 描述           |
| ------ | -------------- |
| next() | 切换到下一面板 |
| prev() | 切换到上一面板 |
