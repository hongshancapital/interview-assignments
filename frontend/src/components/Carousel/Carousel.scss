.carousel-viewport {
    overflow: hidden;
    width: 100%;
    height: 100%;
    position: relative;
    .carousel-wrapper{
        display: inline-block;
        white-space: nowrap;
        height: 100%;
        width: 100%;
        .carousel-item{
            display: inline-block;
            flex: 1;
            width: 100vw;
            height: 100%;
         }
    }
    .carousel-indicator-wrapper{
        position: absolute;
        width: 100%;
        height: 3px;
        bottom: 5%;
        &.hide{
            display: none;
        }
    }
}
