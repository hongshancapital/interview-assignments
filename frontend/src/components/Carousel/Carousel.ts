
import EventEmitter from "../../utils/EventEmitter";
import {Options} from "./Options";

export default class Carousel extends EventEmitter {
  // 配置项
  private readonly options: Options;

  // 容器
  private readonly el: HTMLElement;
  private slideWrapEl: HTMLElement;
  
  private pagenationWrapEl: HTMLElement;
  private slides: NodeListOf<Element>;
  
  // 容器宽度
  private readonly elWidth: number;

  private activeIndex = 0;

  // 是否正在过度
  private isTransiting = false;
  
  private timer: any;

  constructor(el: HTMLElement | string, options?: Partial<Options>) {
    super(['transitionEnd']);
    if (!el) return;
    this.el = (typeof el === 'string' ? document.querySelector(el) : el) as HTMLElement;
    this.elWidth = this.el.offsetWidth;
    this.slideWrapEl = this.el.querySelector('.carousel-wrap') as HTMLElement;

    // 合并选项
    this.options = new Options().merge(options);
    this.init();
  }

  private init() {
    this.initPagenation();
    this.el.appendChild(this.pagenationWrapEl);
    // 初始化事件
    this._initEvents();
    this.autoPlay();
  }

  // 初始化分页器
  private initPagenation() {
    this.slides = this.slideWrapEl.querySelectorAll('.carousel-slide');
    const dotNum = this.slides.length;
    let wrapLen = dotNum;

    if (this.options.loop) {
      // 复制首尾两个slide分别放在slideWrapEl的尾首
      const firstNodeCopy = this?.slideWrapEl?.firstElementChild?.cloneNode(true);
      const lastNodeCopy = this?.slideWrapEl?.lastElementChild?.cloneNode(true);
      if(lastNodeCopy) {
        this.slideWrapEl.insertBefore(lastNodeCopy, this.slideWrapEl.firstElementChild);
      }

      if(firstNodeCopy) {
        this.slideWrapEl.appendChild(firstNodeCopy);
      }
      wrapLen += 2;
      this.slideWrapEl.style.left = -this.elWidth + 'px';
    }

    this.slideWrapEl.style.width = wrapLen * 100 + '%';

    // 分页器
    this.pagenationWrapEl = document.createElement('div');
    this.pagenationWrapEl.className = 'carousel-pagination-wrap';
    
    let iEl:HTMLElement;

    for (let a = 0; a < dotNum; a++) {
      iEl = document.createElement('div');
      iEl.className = (a === this.activeIndex ? 'progress-active progress' : 'progress');
      this.pagenationWrapEl.appendChild(iEl);
    }
  }


  // 绑定事件
  private _initEvents() {
    const pagenations = this.pagenationWrapEl.children;
   
    for(let a = 0; a < pagenations.length; a++) {
      pagenations[a].addEventListener('click', () => this.navigation(a));
    }

    this.slideWrapEl.addEventListener('webkitTransitionEnd', this.onTransitionEnd.bind(this));
    // this.el.addEventListener('mouseenter', this.clearInterval.bind(this));
    this.el.addEventListener('mouseleave', this.autoPlay.bind(this));
    this.emitEvent('onInit');
  }

  private go(dir: number) {
    if (this.isTransiting) return;
    this.isTransiting = true;
    this.slideWrapEl.style.transitionDuration = this.options.speed + 's';
    const len = this.slides.length;
    const index = dir > 0 ? this.activeIndex + 1 : this.activeIndex - 1;
    this.activeIndex = (index + len) % len;
      if (this.options.loop) {
        this.loop(dir);
      }else{
        this.slideWrapEl.style.left = -this.activeIndex * this.elWidth + 'px';
      }
    this.updateDots(this.activeIndex);
  }
  

  private loop(dir: number) {
    const oldLeft = this.getLeft(this.slideWrapEl);
    
    // 这里可以利用下dir的正负
    // const newLeft = dir > 0 ? oldLeft - this.elWidth : oldLeft + this.elWidth;
    const newLeft = oldLeft + this.elWidth * -dir;
    this.slideWrapEl.style.left = newLeft + 'px';
  }

  // 点击圆点导航
  private navigation(index: number) {
    if (this.activeIndex !== index) {
      this.slideWrapEl.style.transitionDuration = this.options.speed + 's';
      this.activeIndex = index;
      this.slideWrapEl.style.left = -(this.activeIndex + 1) * this.elWidth + 'px';
      this.updateDots(index);
    }
  }

  // 底部圆点
  private updateDots(index: number) {
    // console.log(index);
    const dots = this.pagenationWrapEl.childNodes;
    for (let a = 0; a < dots.length; a++) {
      (dots[a] as HTMLElement).classList.remove('progress-active');
      while (dots[a].firstChild) {
        dots[a].removeChild(dots[a].firstChild as HTMLElement);
      }
    }
    (dots[index] as HTMLElement).classList.add('progress-active');
    
    let progressChild:HTMLElement;
    progressChild = document.createElement('div');
    progressChild.className="progress-child";
    dots[index].appendChild(progressChild);
  }

  private onTransitionEnd() {
    // console.log('end');
    if (this.options.loop) {
      const left = this.getLeft(this.slideWrapEl);
      if (left >= 0) {
        this.slideWrapEl.style.transitionDuration = '0s';
        this.slideWrapEl.style.left = '-4000px';
        // console.log('跳最后一个');
      }else if (left <= -(this.slides.length + 1) * this.elWidth) {
        this.slideWrapEl.style.transitionDuration = '0s';
        this.slideWrapEl.style.left = '-800px';
        // console.log('跳第一个');
      }
    }
    this.isTransiting = false;
    
    // 发射选中事件，响应DatePicker.on方法
    this.trigger('transitionEnd', this.activeIndex);
    
    this.emitEvent('onTransitionEnd', this.activeIndex);
  }
  
  private autoPlay() {
    if (this.options.autoplay) {
      this.timer = setInterval(() => this.go(1), this.options.delay);
    }
  }
  
  private clearInterval() {
    if (this.timer) {
      clearInterval(this.timer);
      this.timer = null;
    }
  }

  private getLeft(el: HTMLElement): number{
    return Number(el.style.left.slice(0, -2));
  }

  // 发射自定义事件
  private emitEvent(type: string, ...args: any[]) {
    if (this.options[type]) {
      this.options[type](args);
    }
  }

}