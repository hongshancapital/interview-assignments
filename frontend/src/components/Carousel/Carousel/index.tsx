import React, { useEffect, useMemo, useState } from 'react';
import { CarouselItem, CarouselItemProps } from '../CarouselItem';
import { Dots } from '../Dots';
import style from './index.module.scss';

export interface CarouselProps {
  interval?: number;
  showDots?: boolean;
  data: CarouselItemProps[];
}

export function Carousel(props: CarouselProps) {
  const { showDots, data, interval = 5000 } = props;

  const [current, setCurrent] = useState<number>(-1);

  // 组件挂载之后再设置 current
  useEffect(() => {
    if (data.length) {
      setCurrent(1);
    }
  }, [data]);

  useEffect(() => {
    const timer = setTimeout(() => {
      const now = current >= data.length ? 1 : current + 1;
      setCurrent(now);
    }, interval);

    return () => clearTimeout(timer);
  }, [current, data, interval]);

  const width = useMemo(() => `${data.length * 100}%`, [data]);

  const left = useMemo(() => `-${(current - 1) * 100}%`, [current]);

  return (
    <div className={style['my-carousel']}>
      <div
        className={style.list}
        style={{ width, left, transition: `300ms linear` }}
      >
        {data.map((item) => (
          <CarouselItem {...item} key={item.id} />
        ))}
      </div>

      {showDots && data.length > 1 && (
        <Dots
          setCurrent={setCurrent}
          current={current}
          interval={interval}
          count={data.length}
          data={data}
        />
      )}
    </div>
  );
}
