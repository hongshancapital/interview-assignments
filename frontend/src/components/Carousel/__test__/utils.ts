import { render as ORender, RenderOptions } from '@testing-library/react';

export const render = (ui: React.ReactElement, options?: RenderOptions) => {
  const wrapper = {
    ...ORender(ui, {
      // container: document.body,
      ...options,
    }),
  } as ReturnType<typeof ORender> & {
    find: <E extends HTMLElement | SVGElement>(selector: string) => NodeListOf<E>;
    querySelector: <T extends HTMLElement | SVGElement>(selector: string) => T | null;
  };

  wrapper.find = (selector) => {
    return document.querySelectorAll(selector);
  };
  wrapper.querySelector= (selector) => {
    return document.querySelector(selector);
  };


  return wrapper;
};