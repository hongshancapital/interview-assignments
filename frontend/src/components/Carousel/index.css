html,body,#root{
    width: 100%;
    height: 100%;
}
.Carousel{
    width: 1000px;
    height: 600px;
    position: absolute;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    margin: auto;
    overflow: hidden;
}

.box{
    display:flex;
    position: absolute;
    left: 0;
    top: 0;
    transition:left .5s;
   
}

.content{
    width: 100%;
    height: 100%;
    position: relative;
}
img{
    width: 1000px;
    height: 600px;
}
.pointBox{
    position: absolute;
    left: 0;
    right: 0;
    bottom: 20px;
    display: flex;
    justify-content: center;
    align-items: center;
   
}
.point{
    width: 80px;
    height: 3px;
    background-color: rgb(114, 112, 112);
    margin-right: 3px;
}
.pointInner{
    background-color: rgb(223, 213, 213);
    width:0;
    height: 100%;
    
}
.title{
    position: absolute;
    left: 0;
    right: 0;
    top: 170px;
    color: #fff;
    font-size: 16px;
}
.titleOther{
    color: #000;
}
.titleName{
    font-size: 50px;
}
