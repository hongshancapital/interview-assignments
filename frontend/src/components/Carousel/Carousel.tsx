// <!---------------------------
// Name: Carousel
// File: carousel
// -----------------------------
// Author: YuCheng
// Telphone: 13212707252
// Email: yu656@qq.com
// Data:   1/10/2023, 4:51:59 PM
// ---------------------------->
import React, { useEffect, useRef, useState } from "react";
import "./Carousel.css";

interface props {
  speed?: number;
}
const Carousel = (props: props) => {
  const wrapperRef = useRef<any>();
  const [currentPage, setCurrentPage] = useState<number>(1); //当前显示页
  const [translateX, setTranslateX] = useState<number>(0); //偏移量
  const [viewWidth, setViewWidth] = useState<number>(0); //窗口宽度
  const [childNodes, setChildNodes] = useState<Array<number>>([]); //页码数组
  const [timer, setTimer] = useState<NodeJS.Timeout>(); //定时器
  const [hovered, setHovered] = useState<boolean>(false); //鼠标移入移出状态

  useEffect(() => {
    handleResize();
  }, [wrapperRef]);

  // 定时切换，设置当前页
  useEffect(() => {
    //鼠标移入状态时不启用定时器
    if (!hovered) {
      const timer = setTimeout(() => {
        currentPage === 3 || currentPage < 1
          ? setCurrentPage(1)
          : setCurrentPage(currentPage + 1);
      }, props.speed || 5000);
      setTimer(timer);
    }
  }, [currentPage]);

  // 根据当前页设置偏移量
  useEffect(() => {
    currentPage <= 1
      ? setTranslateX(0)
      : setTranslateX(-(viewWidth * (currentPage - 1)));
  }, [currentPage]);

  useEffect(() => {
    // 监听窗口变化
    window.addEventListener("resize", handleResize);
    // 销毁监听
    return () => window.removeEventListener("resize", handleResize);
  }, [wrapperRef]);

  // 窗口大小改变重新赋值方法
  const handleResize = () => {
    let nodeArr: Array<number> = [];
    wrapperRef?.current?.firstChild?.childNodes.forEach(
      (item: any, index: number) => {
        nodeArr.push(index);
      }
    );
    setChildNodes(nodeArr);
    setViewWidth(wrapperRef?.current?.clientWidth);
  };

  //鼠标移入切换到指定页
  const mouseEnter = (index: number) => {
    setHovered(true);
    setCurrentPage(index + 1);
    clearTimeout(timer);
  };

  //鼠标移出重置为定时切换
  const mouseLeave = (index: number) => {
    setHovered(false);
    setCurrentPage(1);
  };

  //移动端左右滑动切换--滑动开始事件
  let startX: number, endX: number;
  const handelTouchStart = (e: any) => {
    startX = e.touches[0].clientX;
  };
  //移动端左右滑动切换--滑动平移事件
  const handelTouchMove = (e: any) => {
    endX = e.touches[0].clientX;
  };

  //移动端左右滑动切换--滑动结束事件
  const handelTouchend = (e: any) => {
    if (startX - endX > 40) {
      setHovered(true);
      setCurrentPage(currentPage + 1);
      clearTimeout(timer);
    } else if (endX - startX > 40) {
      setHovered(true);
      setCurrentPage(currentPage - 1);
      clearTimeout(timer);
    }
  };

  //监听左右滑动切换
  useEffect(() => {
    wrapperRef?.current?.addEventListener("touchstart", handelTouchStart,{passive: true});
    wrapperRef?.current?.addEventListener("touchmove", handelTouchMove,{passive: true});
    wrapperRef?.current?.addEventListener("touchend", handelTouchend,{passive: true});
    return () => {
      wrapperRef?.current?.removeEventListener("touchstart", handelTouchStart);
      wrapperRef?.current?.removeEventListener("touchMove", handelTouchMove);
      wrapperRef?.current?.removeEventListener("touchend", handelTouchend);
    };
  }, [currentPage]);

  return (
    <div className="wrapper" ref={wrapperRef}>
      <div
        className="imgList"
        style={{ transform: `translateX(${translateX}px)` }}
      >
        <div
          className="imgItem"
          style={{
            backgroundImage: `url(${require("../../assets/iphone.png")})`,
            backgroundColor: "#111111",
            color: "#ffffff",
          }}
        >
          <div className="imgTitle">xPhone</div>
          <div className="imgDes">
            Lots to love.Less to spend.
            <br />
            Strting at $399.
          </div>
        </div>
        <div
          className="imgItem"
          style={{
            backgroundImage: `url(${require("../../assets/tablet.png")})`,
            backgroundColor: "#fafafa",
            color: "#000000",
          }}
        >
          <div className="imgTitle">Tablet</div>
          <div className="imgDes">Just the right amount of everything.</div>
        </div>
        <div
          className="imgItem"
          style={{
            backgroundImage: `url(${require("../../assets/airpods.png")})`,
            backgroundColor: "#f1f1f3",
            color: "#000000",
          }}
        >
          <div className="imgTitle">
            Buy a Tablet or xPhone for college.
            <br /> Get airPods.
          </div>
          <div className="imgDes"></div>
        </div>
      </div>
      <div className="dotList">
        <>
          {childNodes.map((item: number, index: number) => {
            return (
              <div
                className="dotItem"
                key={index}
                onMouseEnter={() => mouseEnter(index)}
                onMouseLeave={() => mouseLeave(index)}
              >
                {currentPage - 1 === index ? (
                  <div
                    className="inner line"
                    style={{
                      animation: `progress ${props.speed || 5000}ms  1 linear`,
                    }}
                  ></div>
                ) : (
                  ""
                )}
              </div>
            );
          })}
        </>
      </div>
    </div>
  );
};
export default Carousel;
