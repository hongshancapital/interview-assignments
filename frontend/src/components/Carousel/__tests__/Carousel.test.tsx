/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import Carousel, { CarouselRef } from '../Carousel';
import { waitFakeTimer } from '../../../tests/utils';
import '@testing-library/jest-dom/extend-expect';

describe('Carousel', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('should render Carousel component', () => {
    const { container } = render(
      <Carousel>
        <div />
      </Carousel>
    );
    expect(container).toBeInTheDocument();
  });

  it('should render the correct number of indicators', () => {
    const { container, rerender } = render(<Carousel />);
    expect(container.querySelectorAll('.indicator').length).toBe(0);

    // Update children
    rerender(
      <Carousel>
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Carousel>
    );
    expect(container.querySelectorAll('.indicator').length).toBe(3);
  });

  it('should active slider when children change', () => {
    const { rerender, container } = render(<Carousel />);
    expect(container.querySelector('.slider-active')).toBeFalsy();

    // Update children
    rerender(
      <Carousel>
        <div />
      </Carousel>
    );
    expect(container.querySelector('.slider-active')).toBeTruthy();
  });

  it('should have goTo, goPrev and goNext function and work normally', async () => {
    const ref = React.createRef<CarouselRef>();
    render(
      <Carousel ref={ref}>
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Carousel>
    );

    const { goTo, goPrev, goNext } = ref.current || {};

    expect(typeof goTo).toBe('function');
    expect(typeof goPrev).toBe('function');
    expect(typeof goNext).toBe('function');

    act(() => {
      ref.current?.goTo(2);
    });
    await waitFakeTimer();
    expect(ref.current?.activeIndex).toBe(2);
    act(() => {
      ref.current?.goPrev();
    });
    await waitFakeTimer();
    expect(ref.current?.activeIndex).toBe(1);
    act(() => {
      ref.current?.goNext();
    });
    await waitFakeTimer();
    expect(ref.current?.activeIndex).toBe(2);
  });

  it('should have start and stop function and work normally', async () => {
    const ref = React.createRef<CarouselRef>();
    render(
      <Carousel ref={ref}>
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Carousel>
    );

    const { start, stop } = ref.current || {};

    expect(typeof start).toBe('function');
    expect(typeof stop).toBe('function');

    expect(ref.current?.activeIndex).toBe(0);

    // wait for animation to be finished
    await waitFakeTimer();
    await waitFakeTimer();
    expect(ref.current?.activeIndex).toBe(1);

    ref.current?.stop();
    await waitFakeTimer();
    expect(ref.current?.activeIndex).toBe(1);

    ref.current?.start();
    expect(ref.current?.activeIndex).toBe(1);

    // wait for animation to be finished
    await waitFakeTimer();
    await waitFakeTimer();
    expect(ref.current?.activeIndex).toBe(2);
  });

  it('should active corresponding slider and indicator when the active index changes', async () => {
    const ref = React.createRef<CarouselRef>();

    // autoplay
    const { container, rerender } = render(
      <Carousel ref={ref}>
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Carousel>
    );

    expect(container.querySelectorAll('.slider')[0]).toHaveClass(
      'slider-active'
    );
    expect(container.querySelectorAll('.indicator')[0]).toHaveClass(
      'is-active-anim'
    );

    act(() => ref.current?.goTo(1));
    await waitFakeTimer();
    expect(container.querySelectorAll('.slider')[1]).toHaveClass(
      'slider-active'
    );
    expect(container.querySelectorAll('.indicator')[1]).toHaveClass(
      'is-active-anim'
    );

    // not autoplay
    rerender(
      <Carousel autoplay={false} ref={ref}>
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Carousel>
    );

    act(() => ref.current?.goTo(0));
    await waitFakeTimer();
    expect(container.querySelectorAll('.slider')[0]).toHaveClass(
      'slider-active'
    );
    expect(container.querySelectorAll('.indicator')[0]).toHaveClass(
      'is-active'
    );

    act(() => ref.current?.goTo(1));
    await waitFakeTimer();
    expect(container.querySelectorAll('.slider')[1]).toHaveClass(
      'slider-active'
    );
    expect(container.querySelectorAll('.indicator')[1]).toHaveClass(
      'is-active'
    );
  });
});
