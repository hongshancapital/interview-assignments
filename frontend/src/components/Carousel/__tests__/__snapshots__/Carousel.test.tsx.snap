// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="carousel-wrap"
    style="background-color: cyan;"
  >
    <div
      class="carousel-list"
      style="transform: translateX(calc(-100% * 0));"
    >
      <div
        class="carousel-item active"
      >
        <div
          class="carousel-item__title"
        >
          title1
        </div>
        <div
          class="carousel-item__content"
        >
          content1
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <div
          class="carousel-item__title"
        >
          title2
        </div>
        <div
          class="carousel-item__content"
        >
          content2
        </div>
      </div>
    </div>
    <div
      class="carousel-indicator"
    >
      <div
        class="carousel-indicator__item active"
      >
        <div
          class="carousel-indicator__timer"
          style="animation-duration: 2000ms;"
        />
      </div>
      <div
        class="carousel-indicator__item"
      />
    </div>
    <div
      class="carousel-nav-bar__prev"
    >
      &lt;
    </div>
    <div
      class="carousel-nav-bar__next"
    >
      &gt;
    </div>
  </div>
</DocumentFragment>
`;
